lib/services/access_request.go

```diff
@@ -1151,7 +1153,13 @@ func (m *RequestValidator) calculateMaxAccessDuration(req types.AccessRequest) (
 	}
 
 	maxDuration := maxDurationTime.Sub(req.GetCreationTime())
-	if maxDuration < 0 {
+
+	// For dry run requests, the max_duration is set to 7 days.
+	// This prevents the time drift that can occur as the value is set on the client side.
+	// TODO(jakule): Replace with MaxAccessDuration that is a duration (5h, 4d etc), and not a point in time.
+	if req.GetDryRun() {
+		maxDuration = maxAccessDuration - 10
```

Comment:
Who/what sends the dry run requests? Wouldn't a more straightforward fix be to just not set it to max 7 days for those?

Also, what are dry run requests and why does it affect only those? A user can also set it to 7 days, no?

Reply:
WebUI needs to know the max duration to show the correct range in the UI. There is no way to know the range without checking all the roles, so we're sending a dry run request and using the max duration to display the correct value in the UI.
Setting this value always to max (7d) would be confusing when you request 7d and the created request will allow only for 4d (shorten by role limit).

Reply:
That makes sense, my point was that if a user's roleset allows 7d requests, then they can also be affected by the time drift issues, no? So it's not just dry run requests. Unless I'm missing something.

Reply:
In the current flow, it won't. The flow is:
```
dry request -> max value is returned -> max value is used to make the "real" request (unless changed to a lower value)
```
This is still not an ideal solution, we should be sending the duration `4d` for example instead of a timestamp, but I think that the current fix is a good workaround to unblock affected people. Later we can change the API to make it less tricky.

Reply:
Oh so the "max duration" in the current implementation will basically always be 7 days - 10 seconds maximum? I understand now but yeah the flow is not clear at all from the current comment, I would consider expanding the comment with the same explanation otherwise future readers will get confused like me.